astropy/units/format/utils.py

```diff
@@ -245,3 +253,12 @@ def unit_deprecation_warning(
     if decomposed is not None:
         message += f" Suggested: {decomposed}."
     warnings.warn(message, UnitsWarning)
+
+
+def _get_non_keyword_units(
```

Comment:
I don't think this should be underscored, since this function is used outside this module (which itself is private under the admittedly poorly documented rules - but nothing in `utils` is `__all__` of this module's `__init__.py`.

Reply:
Even though the Python language does not distinguish between public and private API, the distinction is still useful. Python community has therefore come up with conventions, but because there is no enforcement by the language itself then what has happened is that there are three different common conventions. The way I see it the point of the public API APE proposal is that we should try to follow all three conventions because that's least confusing for both developers and for users (it doesn't really matter which convention someone follows if they all lead to the same conclusion). It might be the case that rewriting existing code is a lot of work, but with brand new private functions there is little reason not to use underscores.

Reply:
For a function, I thought it was quite clear that if it is private, it is not going to be used outside its own module/file, as is indeed the case for `_try_decomposed`, the one function in this module with an underscore, i.e., one should be able to rename it without worrying about use elsewhere (within astropy).

More generally, best to follow the style of the existing code; as is, that has a somewhat clear logic (to me at least, but in this case that counts...).

Reply:
The way I understand it everything we have in our code is either public (we need to be careful not to break backwards compatibility) or private (we can change or remove without notice). I don't see why we would want to complicate things beyond that by using underscores to indicate which files something is used in, but that's what the public API APE is about, so I've removed the underscores here for now.